pkg/auth/azure.go

```diff
@@ -122,3 +145,80 @@ func (a *Azure) ValidateUser(certificates map[string]CertificateList, token stri
 		Expires: time.Unix(int64(claims["exp"].(float64)), 0),
 	}, nil
 }
+
+func (a *Azure) GroupsForUser(token, email string) (Groups, error) {
+	bearerToken, err := a.getBearerTokenOnBehalfOfUser(token)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodGet, AzureGraphMemberOfEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %v", bearerToken))
+	response, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var body []byte
+	response.Body.Read(body)
+
+	var memberOfResponse MemberOfResponse
+	if err := json.NewDecoder(response.Body).Decode(&memberOfResponse); err != nil {
+		return nil, err
+	}
+
+	var groups Groups
```

Comment:
Why make a new list of groups? Isn't the respons enough?

Reply:
it is for the filtering logic, which may be removed